std_execution.bs

```diff
@@ -3002,19 +3002,7 @@ Insert this section as a new subclause, between Searchers <b>[func.search]</b> a
 <ins>
 <blockquote>
 
-1. The name `std::tag_invoke` denotes a customization point object. Its type is an unspecified class type equivalent to:
-
-    <pre highlight="c++">
-    struct <i>tag-invoke</i>::<i>tag</i> {
-      template &lt;class Tag, class... Args>
-          requires tag_invocable&lt;Tag, Args...>
-        constexpr auto operator()(Tag tag, Args&&... args) const
-          noexcept(nothrow_tag_invocable&lt;Tag, Args...>)
-          -> tag_invoke_result_t&lt;Tag, Args...> {
-          return tag_invoke(auto(tag), (Args&&) args...);
-        }
-    };
-    </pre>
+1. The name `std::tag_invoke` denotes a customization point object [customization.point.object]. Given subexpressions `T` and `A...` the expression `tag_invoke(T, A...)` is expression-equivalent [defns.expression-equivalent] to `tag_invoke(T, A...)` if it is a well-formed expression with overload resolution performed in a context that does not include the declaration of `std::tag_invoke`. Otherwise, `tag_invoke(T, A...)` is ill-formed.
```

Comment:
Cribbing from the `ranges::begin` wording and adding the `void tag_invoke();` declaration to avoid the degenerate case where phase 1 name lookup on `tag_invoke` finds an object instead of a function:

```suggestion
1. Given subexpressions `T` and `A...`, if `tag_invoke(T, A...)` is a valid expression with overload resolution performed in a context in which unqualified lookup for `tag_invoke` finds only the declaration

    ```c++
    void tag_invoke();
    ```

then `std::tag_invoke(T, A...)` is expression-equivalent [defns.expression-equivalent] to `tag_invoke(T, A...)` with overload resolution performed in the above context.

2. [Note: Diagnosable ill-formed cases above result in substitution failure when `std::tag_invoke(T, A...)` appears in the immediate context of a template instantiation. —end note]
```